### ISSUE TITLE
Question: How to send score from my game?

### ISSUE THREAD

**Issue by easchimidt** [telegraf/telegraf#47]
Hello all,

I am using Telegraf with games. It's working very well, the game is opening without problems.

Now I want to sent user scores to Telegram. But my game has no communication at all with my bot. How do you recommend sending scores to Telegram, as I cannot use telegraf's implementation.

Thanks in advance not only for answering my question, but also for building Telegraf and sharing it with community. Cheers!

**Comment by dotcypress** [comment]
Hey.

You might see implementation example in [telegraf/microgames repo](https://github.com/telegraf/microgames)

**Comment by easchimidt** [comment]
Hey dotcypress,

Thanks for your fast answer and for sharing this new repo. I couldn't get back to my game until today. So, I am finally deep diving into it.

Cheers.

**Comment by easchimidt** [comment]
Hey dotcypress, can you help me?

I have a server (nginx) serving my games in production (in development each game has it own server).

I have created the saveScore function, using XMLHTTPRequest as your example. I am passing as parameters the info requested by telegram (chatid, userid, messageid), and can get it inside the game and build the req message with them.

But, I still couldn't get to the end, sending to telegram with the function routed in /score.

My route is inside telegraf server, but something is missing, as the url is not being served.

Here are my doubts: 

1. Is telegraf a server? Is it listening to any ports? If so, could I send saveScore from inside the game to the internal telegraf url + /saveScore?
2. If not possible, do you have any suggestions? Start a telegraf instance inside each game? (I will have more games in the future, and would like to have only one Bot (Telegraf) handling them all, not one per game).

Well, thanks for your time. Hope anybody can help :)

**Comment by dotcypress** [comment]
Hey @easchimidt

1. Is telegraf a server?

Yes, if you use [`startWebhook `](https://github.com/telegraf/telegraf/blob/develop/lib/telegraf.js#L61).

You can add optional handler to stick with custom routes, static serving, etc:

`startWebhook (path, tlsOptions, port, host, cb)` where `cb` is http handler function:

```js
const cb = (req, res) => {
  res.end()
}
```
Also checkout [simple implementation with express](https://github.com/telegraf/microgames/blob/master/index.js#L58)

The other way is using ('webhookCallback')[http://telegraf.js.org/telegraf.html#webhookcallback] with existing http/express/koa server:

```js
https.createServer(options, bot.webhookCallback()).listen(8000);
```

**Comment by easchimidt** [comment]
Thanks, @dotcypress 

I thought startWebhook was used only to receive calls from Telegram. Is it advisable to use it to receive data from player, too? Is it safe, as clients have access to JS, and can see the url path of webhook used to save score? It looks like a possible security flaw.

I was looking into this [simple implementation with express](https://github.com/telegraf/microgames/blob/master/index.js#L58) to try solving my problem - and this is exactly the point I am stuck - creating a router that handles a url to send new score from user to telegram.

Do you think it is better to create a server inside my telegraf, even I have nginx behind? Then it can become the "express server used to telegram". Gonna try it out.

Thanks, dotcypress :D

**Comment by easchimidt** [comment]
Hey guys,

I finally found a solution... Sharing here as it could help someone...

In my nginx, I created a location to /my-game-score, proxying to a port in localhost that is running my telegraf/server.

In my telegraf, I created some rules similar to [microgames](https://github.com/telegraf/microgames/blob/master/index.js) sample done by dotcypress.

Before I send gameUrl to the user in Telegram, I append userId/chatId/message/iMessage to the url, as they will be used to save score at the end.

When game finishes, I create a XMLHTTRequest, and post an object with the info above plus the score. In my /my-game-score route I do the checking and send to Telegram using Telegraf.Telegram.

Thanks for all the help, dotcypress, and for developing and sharing Telegraf!

In my game, a create a XMLHTTPRequest
